ngReact.js

```diff
@@ -234,24 +264,24 @@
           // if props is not defined, fall back to use the React component's propTypes if present
           props = props || Object.keys(reactComponent.propTypes || {});
 
-          var propNames = props.map(getPropName);
-
           // for each of the properties, get their scope value and set it to scope.props
           var renderMyComponent = function() {
-            var props = {};
-            propNames.forEach(function(propName) {
-              props[propName] = scope.$eval(attrs[propName]);
+            var scopeProps = {}, config = {};
+            props.forEach(function(prop) {
+              var propName = getPropName(prop);
+              scopeProps[propName] = scope.$eval(findAttribute(attrs, propName));
+              config[propName] = getPropConfig(prop);
             });
-            props = applyFunctions(props, scope);
-            props = angular.extend({}, props, injectableProps);
-            renderComponent(reactComponent, props, scope, elem);
+            scopeProps = applyFunctions(scopeProps, scope, config);
+            scopeProps = angular.extend({}, scopeProps, injectableProps);
+            renderComponent(reactComponent, scopeProps, scope, elem);
           };
 
           // watch each property name and trigger an update whenever something changes,
           // to update scope.props with new values
           var propExpressions = props.map(function(prop){
             return (Array.isArray(prop)) ?
-              [attrs[prop[0]], prop[1]] :
+              [attrs[getPropExpression(prop)], getPropConfig(prop)] :
```

Comment:
`getPropName` and `getPropExpression` do functionally the same thing, so this will work either way. But for what it's worth, you're actually getting the prop _name_ when calling `getPropExpression` here. Using `getPropName` might be slightly clearer.